### ISSUE TITLE
micro benchmark - build

### ISSUE THREAD

**Issue by yishayahu** [readablesystems/sto#6]
We are trying to build and run the micro benchmark.
We get an error in the compilation - no "select_row" is defined for ordered_index.
Indeed, looking at the "DB_oindex.hh" we see that this function implementation is under "if 0".
Would you please explain to us if and how to fix that?
Thanks, 
Shaya

**Comment by wqian94** [comment]
For the purposes of testing our timestamp splitting via split versions implementation, we replaced `select_row` with `select_split_row`. If you do not intend to use this feature, `select_row` should work just fine if you uncomment it.

**Comment by yishayahu** [comment]
we tried to uncomment select_row, it is not working for us.
Thanks

**Comment by wqian94** [comment]
Is this after uncommenting both `select_row` overrides? One of them calls the other, so both should be uncommented.

If you're using commit `013f5e`, these are lines 118-132 and 150-195.

**Comment by yishayahu** [comment]
yes, we tried to uncomment both of them, and that's the commit we are using.

**Comment by wqian94** [comment]
Can you please send the error trace that you are seeing?

**Comment by yishayahu** [comment]
benchmark/DB_oindex.hh:129:66: error: could not convert ‘{true, false, 0, <expression error>}’ from ‘<brace-enclosed initializer list>’ to ‘bench::ordered_index<ubench::MasstreeIntKey<long unsigned int>, eight_version_row, db_params::db_swiss_params>::sel_return_type {aka std::tuple<bool, bool, long unsigned int, const eight_version_row*>}’
             return {true, false, 0, UniRecordAccessor<V>(nullptr)};

**Comment by wqian94** [comment]
Ah -- do you have `UniRecordAccessor` defined for `eight_version_row`? `UniRecordAccessor` should be auto-coerced into the appropriate pointer type.

**Comment by yishayahu** [comment]
no, how can I define it?

**Comment by wqian94** [comment]
It's a little roundabout, but if you look at lines 4-236 in `benchmark/tpcc_split_params_default.hh`, that's an example of the specialized template classes for `warehouse_value` from TPC-C. In particular, the UniRecordAccessor and SplitRecordAccessor classes are essentially the same for this case.

If you're looking to use the timestamp splitting feature, you'll want to look at `benchmark/tpcc_split_params_ts.hh` instead.
